bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/SemanticTagRegistry.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.semantics;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.common.registry.Registry;
+
+/**
+ * {@link SemanticTagRegistry} tracks all {@link SemanticTag}s from different {@link SemanticTagProvider}s
+ * and provides access to them.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public interface SemanticTagRegistry extends Registry<SemanticTag, String> {
+
+    /**
+     * Retrieves the class for a given id.
+     *
+     * @param tagId the id of the tag. The id can be fully qualified (e.g. "Location_Room_Bedroom") or a segment, if
+     *            this uniquely identifies the tag
+     *            (e.g. "Bedroom").
+     * @return the class for the id or null, if non exists.
+     */
+    @Nullable
+    Class<? extends Tag> getTagClassById(String tagId);
+
+    /**
+     * Builds the fully qualified id for a semantic tag class.
+     *
+     * @param tag the semantic tag class
+     * @return the fully qualified id
+     */
+    String buildId(Class<? extends Tag> tag);
```

Comment:
Why is this part of the registry, shouldn't it be part of `SemanticTag`?

Reply:
Not in `SemanticTag` but I agree we should probably not expose this method in the registry interface.
This method is mainly required by the semantic metadata provider.
I propose to create a static method.
It is in the registry implementation because the idea is to have all logic with ids in that place.

Reply:
Done